### DISCUSSION TITLE
Is there any to get getServerSideProps functionality in App Router?

### DISCUSSION THREAD

**Discussion by pranair** [vercel/next.js#59188] [App Router] [answered]
Hi, I have a following scenario:
- Client sends a SAML response to a NextJS page
- NextJS sends the SAML response to the backend and receives a JWT if SAML is valid.
- Client sees a loading page all the while this happens

This used to be possible with getServerSideProps, which was how it used to be done. However, after migrating too App Router it seems a different approach is needed. 

Is it possible to send a POST request to a NextJS page?

**Comment by boncz92** [comment]
I would just send the POST request directly to an API Route and then from that API route after validating the SAML response and issueing a JWT cookie redirect them to whatever page they were attempting to load.

**Reply by pranair** [reply, marked as answer]
This could have worked, but I was explicitly asked to show a loading page. I ended up using page router for just that page.
